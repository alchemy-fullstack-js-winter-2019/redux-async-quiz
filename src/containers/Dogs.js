import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Image from '../components/Image';
import { connect } from 'react-redux';
import { getDogSelect, isLoading } from '../selectors/dogs';
import { fetchDog } from '../actions/dogs';

class Dogs extends PureComponent {
  static propTypes = {
    dog: PropTypes.object,
    loading: PropTypes.bool,
    fetch: PropTypes.func
  }

  componentDidMount() {
    this.props.fetch();
  }

  render() {
    return (
      <Image 
        dog={this.props.dog}
        loading={this.props.loading}
        fetch={this.props.fetch}
      />
    );
  }
}

const mapStateToProps = state => ({
  dog: getDogSelect(state),
  loading: isLoading(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(fetchDog());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Dogs);

