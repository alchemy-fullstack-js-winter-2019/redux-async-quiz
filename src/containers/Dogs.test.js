import React from 'react';
import Dogs from './Dogs';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Home snapshot', () => {
  it('can match a snapshot', () => {
    const tree = shallow(
      <Dogs />
    );
    expect(tree).toMatchSnapshot();
  });
});
