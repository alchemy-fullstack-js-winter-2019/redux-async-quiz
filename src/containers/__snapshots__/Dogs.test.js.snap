// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot can match a snapshot 1`] = `ShallowWrapper {}`;
